### PR TITLE
implement standard as patterns in mono IR

### PR DESCRIPTION
**Pull request by folkertdev**
the special `[ first, .. as rest ]` list pattern is not yet implemented.

### CODE REVIEW

File: crates/compiler/mono/src/ir.rs [hunk 1]
```diff
@@ -7411,6 +7411,28 @@ fn store_pattern_help<'a>(
...
+            // An identifier in a pattern can define at most one specialization!
+            // Remove any requested specializations for this name now, since this is the definition site.
+            let specialization_symbol = procs
+                .symbol_specializations
+                .remove_single(*symbol)
+                // Can happen when the symbol was never used under this body, and hence has no
+                // requested specialization.
+                .unwrap_or(*symbol);
+
+            substitute_in_exprs(env.arena, &mut stmt, specialization_symbol, outer_symbol);
```
Comment (folkertdev) [inline crates/compiler/mono/src/ir.rs:7432]: @ayazhafiz does this reasoning make sense for the identifier introduced by `as`? it's the same as for normal `Identifier` patterns, which I think is correct.

Comment (ayazhafiz) [reply crates/compiler/mono/src/ir.rs:7432]: Yeah that seems right.
I'm working on weakening let-generalization so when that pans out, we'll be able to rip out all of this (finally)

**Comment by folkertdev** [comment]
this fails on the markdown links but that is not a required CI step

**Review by ayazhafiz — APPROVED** [review]
🔥

**Pull request merged**